tests/integrated/test-dataiterator/runtest

```diff
@@ -6,59 +6,31 @@
 
 from __future__ import print_function
 try:
-  from builtins import str
+    from builtins import str
 except:
-  pass
+    pass
 from boututils.run_wrapper import shell, shell_safe, launch_safe, getmpirun
 from boutdata.collect import collect
 from sys import stdout, exit
 
 MPIRUN=getmpirun()
 
 print("Making DataIterator test")
-s, out = launch_safe("make > make.log",pipe=True)
-if s:
-  print("Failed to run make")
-  exit(s);
+shell_safe("make > make.log",pipe=True)
 
-flags = [""]
-#, "nsys=2", "nsys=5 periodic", "nsys=7 n=10"]
-
-code = 0 # Return code
 for nproc in [1,2,4]:
-  for mproc in [1,2,4,8]:
-    cmd = "./test_dataiterator"
+    for mproc in [1,2,3,4]:
+        cmd = "./test_dataiterator"
+
+        print("   %d processors...." % (nproc))
 
-    print("   %d processors...." % (nproc))
-    r = 0
-    for f in flags:
         stdout.write("\tflags '"+f+"' ... ")
 
         shell("rm data/BOUT.dmp.* 2> err.log")
 
         # Run the case
-        s, out = launch_safe(cmd+" "+f, runcmd=MPIRUN, nproc=nproc, pipe=True)
-        f = open("run.log."+str(nproc)+"."+str(mproc), "w")
-        f.write(out)
-        f.close()
-        r = r + 1
-        if s:
-          #stdout.write(out)
-          code=1
-          print("FAILED")
-        else:
-          
-          # Find out if it worked
-          #allpassed = collect("allpassed", path="data", info=False)
-          #if allpassed:
-          print("PASSED")
-          #else:
-          #  print("FAILED")
-          #  code = 1
-
-if code == 0:
-    print(" => All data iteration tests passed")
-else:
-    print(" => Some failed tests")
+        _, out = launch_safe(cmd+" "+f, runcmd=MPIRUN, nproc=nproc, mthread=mproc pipe=True)
```

Comment:
Missing comma after `mproc`.

I'm not sure about using `launch_safe` here, as it will immediately bail out and not run the further tests. Let's say threading is broken, but MPI works ok -- as it is, it will fail on `nproc=1, mproc=1`, and we won't see that `nproc=2, mproc=1` still works. Obviously, the whole test should fail, but it would be nice to run as many tests as possible to see if it's only a subset that's broken.

Reply:
Thanks - fixed the missing `,`

Now it should runs all tests :+1: